.github/workflows/publish.yml

```diff
@@ -1,39 +1,38 @@
 name: eve-kernel build
+
 on:
+  workflow_dispatch:
   pull_request_review:
-    types: [submitted]
-  pull_request:
-    types: [opened, synchronize, closed]
+    types: [ submitted ]
```

Comment:
I don't understand this.
When someone submits a PR it is ok to go build it (to make sure it compiles) but not push it to docker hub,

However, this doesn't seem to build the PR but what is already in the branch (and then push that).

I think we need to build and push to docker hub from the branch when a pull_requst has been merged and on a push. 
And then add (potentially a separate build_pr.yml) to build the pull request when submitted. 
Or am I missing something?

Reply:
I have modified the actions to checkout the `ref: ${{ github.head_ref }}` - this will ensure we are building the branch as  well as the PR build. 

Hope this resolves the issue.